### PR TITLE
Dynamic NPCs: find a base and travel there.

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Dynamic NPCs: find a base and travel there."```

#### Purpose of change
Dynamic NPCs will now choose a likely safe spot for them to use as a base of operations, and will travel there.

This is the first step in a series of planned features.

in #34055 I added NPC AI flags to all overmap locations that were applicable.
Now I load those into overmap_location, the NPC AI has been tweaked to set the priority the higest to find a "source_of_safety" nd the nearest likely candidate for such, and travel there.

Once they travel there, they will stay there indefinitely ( for the moment ).

Later on what will happen, is that once they have a base, they will then go on looting missions, or foraging or various other tasks, and take that stuff back to their base, build up their base dynamically and organically.

They will merge their dynamic factions with other NPCs if they arrive somewhere and another NPC has already claimed that spot, or they will fight or run away, depending on personalities.

What this currently does however, is at least stop them all rushing towards the nearest city to die, you may now find various dynamic NPCs holed up in various safe-ish spots around the map.

#### Describe the solution
As above

#### Describe alternatives you've considered
Doing the entire plan in one big PR, but Im ( slowly ) learning my lessons about yak-shaving

#### Testing
Spawned dynamic NPC, when I was invisible so the NPC didnt get aggressive or fearful, let them choose a destination after they `decide_needs()` they chose to find a `source_of_safety` I debugged in to show their path on the map, it led to a nearby farm.

I waited until they had travelled there on the overmap.
teleported in, and they were correctly set to a "guard" mission, and talked to me with the same attitude they woulda had anyway.

#### Additional context
N/A

### CODE REVIEW

**Comment by Brytenwalda** [comment]
That’s a big step in NPC development. Congrats!

**Comment by Brytenwalda** [comment]
Also, I assume it only works within the reality bubble, right?

**Comment by davidpwbrown** [comment]
> Also, I assume it only works within the reality bubble, right?

No, NPCs travel on the overmap outside of reality bubble now.
When they get to their destination they stop travelling.
This is where we would then add some code for simulating looting/doing stuff when they get there,
but for now they just stop.

**Comment by Brytenwalda** [comment]
If they could loot and so deprive the player of resources that would be a game-changer. Realism-wise this is one of the main things that are missing IMHO.

**Comment by esotericist** [comment]
> If they could loot and so deprive the player of resources that would be a game-changer. Realism-wise this is one of the main things that are missing IMHO.

We've got a lot of plans to the end of simulating looting both in worldgen and as the game progresses

Basically, long term, we want to set things up so that places "spawn" with all of the stuff they SHOULD have had as of the moment of the cataclysm, so anything that 'should' be there that's missing got taken by somebody.

But one step at a time. :)

File: src/overmap.h [hunk 1]
```diff
@@ -152,6 +152,43 @@ class overmap_special_batch
...
+    { "SOURCE_FORAGE", source_forage },
+    { "SOURCE_FUEL", source_fuel },
+    { "SOURCE_GUN", source_gun },
+    { "SOURCE_LUXURY", source_luxury },
+    { "SOURCE_MEDICINE", source_medicine },
+    { "SOURCE_PEOPLE", source_people },
+    { "SOURCE_SAFETY", source_safety },
+    { "SOURCE_TAILORING", source_tailoring },
+    { "SOURCE_VEHICLES", source_vehicles },
+    { "SOURCE_WEAPON", source_weapon }
```
Comment (Night-Pryanik) [inline src/overmap.h]: Why is there such strange indentation? First 6 lines have several spaces before the closing bracket, but it didn't indent to anything below it.

**Comment by davidpwbrown** [comment]
closing this as its dependent PR #35146 got merged first

**Pull request closed without merging**